ui/components/component-library/modal-overlay/modal-overlay.types.ts

```diff
@@ -1,6 +1,6 @@
-import { BoxProps } from '../../ui/box/box.d';
+import { BoxComponent } from '../box/box.types';
```

Comment:
I think you may want to import the stye utility props and polymorphic as prop
```suggestion
import type {
  StyleUtilityProps,
  PolymorphicComponentPropWithRef,
} from '../box';
```

Reply:
Thank you for your feedback. I will make the necessary improvements and address TypeScript issues in the next commit.